### PR TITLE
RHDEVDOCS-5560: Document how to enable Openshift monitoring stack to …

### PR DESCRIPTION
**Pull request by eromanova97**
• **Aligned team**: Dev Tools
• **Please cherry-pick to**: `gitops-docs-1.10` `gitops-docs-1.11`
• **JIRA issues**: [RHDEVDOCS-5560](https://issues.redhat.com/browse/RHDEVDOCS-5560), [OCPBUGS-21819](https://issues.redhat.com/browse/OCPBUGS-21819)
• **Preview pages**: 

- [Installing Red Hat OpenShift GitOps Operator in web console](https://67805--docspreview.netlify.app/openshift-gitops/latest/installing_gitops/installing-openshift-gitops#installing-gitops-operator-in-web-console_installing-openshift-gitops)

- [Installing Red Hat OpenShift GitOps Operator using CLI](https://67805--docspreview.netlify.app/openshift-gitops/latest/installing_gitops/installing-openshift-gitops#installing-gitops-operator-using-cli_installing-openshift-gitops)

- [Monitoring the GitOps Operator performance](https://67805--docspreview.netlify.app/openshift-gitops/latest/observability/monitoring/monitoring-the-gitops-operator-performance)

• **SME Review**:  @iam-veeramalla, @jaideepr97
• **QE review**: @varshab1210

### CODE REVIEW

File: modules/installing-gitops-operator-in-web-console.adoc
Comment (ekristova) [inline modules/installing-gitops-operator-in-web-console.adoc:28]: @Srivaralakshmi I am not happy with this admonition being placed here as it is not specific  just to the installation in web console. But the introduction to the installation section already has two long admonitions so I do not think it is a good idea to add an another one. 

I wanted to hear your opinion on this. Maybe we could leave it there and also add it in the CLI section? But then we would have two same admonitions in two consecutive sections, which is also not ideal. :/

File: modules/installing-gitops-operator-in-web-console.adoc
Comment (ekristova) [inline modules/installing-gitops-operator-in-web-console.adoc:23]: @Srivaralakshmi as you discovered: 
```
If i am installing gitops 1.9.3 ( any version prior to 1.10), am I supposed to see the `openshift-gitops-operator` 
namespace as the default installation namespace or `openshift-operators`? For ROSA, I am seeing only 
`openshift-operators` as the default installation namespace. Why is there an inconsistency in the experience? In 
the NOTE before the procedure, probably we have to mention this as an admonition for people who are 
upgrading to 1.10 from previous releases.
```
I checked this behavior again. With new versions, ROSA is acting the same/has the same issue. But I found out that the `openshift-gitops-operator` namespace is shown because of the chosen **update channel** being `gitops-1.10`. Once the user chooses an older channel, the namespace is no longer offered.

I found an IMPORTANT admonition here: [https://67805--docspreview.netlify.app/openshift-gitops/latest/installing_gitops/installing-openshift-gitops#prerequisites_install-gitops](https://www.google.com/url?q=https://67805--docspreview.netlify.app/openshift-gitops/latest/installing_gitops/installing-openshift-gitops%23prerequisites_install-gitops&sa=D&source=docs&ust=1700036530095818&usg=AOvVaw0O7dx7OgjEWQBa-WIk7ylx) that specifies the user should use the corresponding channel, so this might be an expected behavior, but still worth pointing out.

File: modules/installing-gitops-operator-using-cli.adoc
Comment (ekristova) [inline modules/installing-gitops-operator-using-cli.adoc:11]: @Srivaralakshmi because of the new installation namespace, this section needed some additional changes. It was not enough to just change the name of the namespace. I tested the procedure to make sure it works. The changes are based on [OCPBUGS-21819](https://issues.redhat.com/browse/OCPBUGS-21819)

File: modules/installing-gitops-operator-using-cli.adoc
Comment (ekristova) [inline modules/installing-gitops-operator-using-cli.adoc:113]: @Srivaralakshmi as we also have a second namespace to check now, I added it as a step as well.

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (ekristova) [inline modules/gitops-accessing-the-gitops-operator-metrics.adoc:28]: In the tool-tip of the web console, we can see the description and type of these metrics as we enter them in the query.

But we are missing that information for `active_argocd_instance_reconciliation_count`. By the definition, it should be a counter but the description is missing in the UI.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@ekristova Great work, Thanks a lot! PTAL at the suggestions and questions.

File: modules/installing-gitops-operator-in-web-console.adoc
Comment (ekristova) [reply modules/installing-gitops-operator-in-web-console.adoc:34]: Hello @Srivaralakshmi I did not include the `cluster-admin` role part because I realized it is in the prerequisites. Should I include it here anyway?

Comment (Srivaralakshmi) [reply modules/installing-gitops-operator-in-web-console.adoc:34]: not required. retain as is. Thanks for the clarification.

File: modules/installing-gitops-operator-in-web-console.adoc
Comment (ekristova) [reply modules/installing-gitops-operator-in-web-console.adoc:32]: Agreed. I'll add it there as well.

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (ekristova) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: @Srivaralakshmi as I put these metrics to the table, I realized that this one metric is obsolete, as we describe its three forms in detail below. The `controller_runtime_reconcile_time_seconds_per_instance` metric cannot be observed in this form, which could only confuse the user, so I decided to remove it from the table and include its three versions explicitly (that is `_bucket` `_count` `_sum`. WDYT?

Comment (Srivaralakshmi) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: Makes sense. Please go ahead. Thank you!

File: modules/installing-gitops-operator-using-cli.adoc
Comment (ekristova) [inline modules/installing-gitops-operator-using-cli.adoc:124]: Hello @jaideepr97, I added `openshift-gitops-dex-server-` as an example pod after the user uses the `oc get pods -n openshift-gitops` command, as I could see that pod when testing this. Is it okay to include?

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (ekristova) [inline observability/monitoring/monitoring-the-gitops-operator-performance.adoc:13]: @jaideepr97 I included the type of metrics (gauge/counter), is that beneficial to the user to know?

File: modules/installing-gitops-operator-using-cli.adoc
Comment (ekristova) [reply modules/installing-gitops-operator-using-cli.adoc:134]: @jaideepr97 is it beneficial to the user to verify with `oc get pods -n openshift-gitops-operator` that those pods in the `openshift-gitops-operator` namespace are running, or is checking pods in `openshift-gitops` namespace enough?

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (ekristova) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:28]: cc: @jaideepr97
Here is a screenshot to see what we mean. As you can see, the `active_argocd_instance_reconciliation_count` description and type is missing.
Screenshot from 2023-11-22 11-30-03

Screenshot from 2023-11-22 11-34-47

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (ekristova) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:25]: cc: @jaideepr97
Also question from Sri: Is this a UI bug? If it is, then when the docs team creates a bug for this, who is the person to whom we can assign this? I will create a bug.

**Review by eromanova97 — COMMENTED** [review]
Hello @jaideepr97 can you PTAL at the PR and give clarification to our questions? Thank you!

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (ekristova) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: cc: @jaideepr97

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (ekristova) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: cc: @jaideepr97

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (iam-veeramalla) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:25]: This is not a bug. The behavior cannot be consistent because the above metrics are for group of instances whereas this metric is for a single Argo CD Instance. 

If you would like to create a bug to discuss further. Please create the bug on the GitOps board and keep it unassigned.

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (iam-veeramalla) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:28]: Please create a bug and assign it to @jaideepr97

File: modules/installing-gitops-operator-using-cli.adoc
Comment (iam-veeramalla) [reply modules/installing-gitops-operator-using-cli.adoc:124]: that's fine.

File: modules/installing-gitops-operator-using-cli.adoc
Comment (iam-veeramalla) [reply modules/installing-gitops-operator-using-cli.adoc:134]: Thanks for adding that @ekristova

**Review by iam-veeramalla — APPROVED** [review]

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (Srivaralakshmi) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:25]: Thanks, @iam-veeramalla for the clarification. No need to create a big then.

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (Srivaralakshmi) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: @iam-veeramalla This query is pending. Will you be able to help or do we need to wait for @jaideepr97?

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (Srivaralakshmi) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc:13]: @ekristova I find it beneficial.

@iam-veeramalla This query is pending. Will you be able to help or do we need to wait for @jaideepr97?

**Comment by ekristova** [comment]
Hello @varshab1210  can you please review this PR? Thank you!

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (jaideepr97) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:28]: we don't do anything different while registering these metrics though
it could be a openshift console UI thing because the length of the metric is too long to display the type as well?
I can investigate

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (jaideepr97) [inline modules/gitops-accessing-the-gitops-operator-metrics.adoc:15]: After doing some testing of my own I think the only restrictions for metrics to show up are that
- the operator must be installed in a namespace starting with `openshfit-` like `openshfit-xyz`
- cluster monitoring should be enabled

File: modules/installing-gitops-operator-using-cli.adoc
Comment (jaideepr97) [reply modules/installing-gitops-operator-using-cli.adoc:134]: @ekristova its always a good idea to verify that the operator pod is running without any issues after a new installation, so I would say its good to have this here

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (jaideepr97) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc:13]: Users would need to know what they mean for it to be useful
Guage = value can go up or down
Counter = value can only go up

I don't really have strong opinions either way, we can leave them in

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (jaideepr97) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: `controller_runtime_reconcile_time_seconds_per_instance_bucket` 
for this metric we define certain buckets that represent time durations such as 0.1s, 0.2s, 0.5s etc 
prometheus observes how long each reconciliation cycle for a given instance takes, and populates the graph. So at the end of say 15 reconcilations, we are able to see how many were completed under 0.2s, how many took longer like 0.5s and so on

`controller_runtime_reconcile_time_seconds_per_instance_count` shows the total count of how many such reconciliations were observed  

``controller_runtime_reconcile_time_seconds_per_instance_sum` shows the total amount of time taken for those 15 reconciliations

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (jaideepr97) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: generally yes

can we reword this as:
`controller_runtime_reconcile_time_seconds_per_instance_count`:  The total number of reconciliation cycles observed for the given instance.
?

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (Srivaralakshmi) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: @jaideepr97 thanks for the explanation. @ekristova PTAL and do the needful. Thank you!

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (Srivaralakshmi) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: @thanks @jaideepr97 , will do.  @ekristova PTAL and do the needful. Thank you!

@jaideepr97 How about the description of ** `controller_runtime_reconcile_time_seconds_per_instance_sum`**? Is the definition accurate? Or should we rewrite this a little better? we need your help.
Does this read correctly? Please let us know.

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (Srivaralakshmi) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc:13]: @ekristova PTAL and do the needful. TY!

File: modules/installing-gitops-operator-using-cli.adoc
Comment (Srivaralakshmi) [reply modules/installing-gitops-operator-using-cli.adoc:134]: Thanks @jaideepr97

File: modules/gitops-accessing-the-gitops-operator-metrics.adoc
Comment (Srivaralakshmi) [reply modules/gitops-accessing-the-gitops-operator-metrics.adoc:28]: Sure, @jaideepr97 . Thanks a lot.

Here is the bug I created: https://issues.redhat.com/browse/GITOPS-3665
PTAL. (FYI: @iam-veeramalla)

File: observability/monitoring/monitoring-the-gitops-operator-performance.adoc
Comment (ekristova) [reply observability/monitoring/monitoring-the-gitops-operator-performance.adoc]: The description of `controller_runtime_reconcile_time_seconds_per_instance_sum` was given here: https://github.com/openshift/openshift-docs/pull/67805#discussion_r1408368461
Thank you!

File: modules/installing-gitops-operator-in-web-console.adoc
Comment (jaideepr97) [inline modules/installing-gitops-operator-in-web-console.adoc:27]: The default namespace changed from `openshift-operators` to `openshift-gitops-operator`, it was never installed in `openshift-gitops` namespace

Also, upgrading operators in a different namespace should not cause any issues. The only known issue was that of an alert being generated from a mismatch in service monitor configuration on upgrade, but that should be fixed in 1.11.0

Comment (ekristova) [reply modules/installing-gitops-operator-in-web-console.adoc:27]: @jaideepr97 I see, I did not even notice that I included `openshift-gitops` instead of `openshift-operators`,  thank you for noticing! Also, since this is being fixed in 1.11 (and 1.10.2 as I see in the Jira that tracks the issue) I think it would be best to just include this as a note:
```For the GitOps version 1.10 and later, the default namespace changed from `openshift-operators` to `openshift-gitops operator`.``` WDYT?

Comment (jaideepr97) [reply modules/installing-gitops-operator-in-web-console.adoc:27]: Sounds good

**Review by jaideepr97 — APPROVED** [review]
LGTM thanks @ekristova

**Comment by ekristova** [comment]
Hello @Srivaralakshmi can you please also take a look at the PR again and approve if everything looks good? Thank you!

**Review by Srivaralakshmi — COMMENTED** [review]
@ekristova Great work! PTAL at my suggestions and rectify. Thank you!

File: modules/installing-gitops-operator-using-cli.adoc
Comment (ekristova) [reply modules/installing-gitops-operator-using-cli.adoc:10]: Oh, thank you for noticing! I missed that, making the changes now

**Comment by ekristova** [comment]
Hello @Srivaralakshmi i made the changes and got the SME and QE approval, so I am pinging you as we discussed, thank you!

**Review by Srivaralakshmi — COMMENTED** [review]
@ekristova Excellent job! Neat PR. LGTM and approved from my side.

As I am new to the peer review squad, I would ask @bmcelvee to give my review the go-ahead. Please wait until Brandi applies the peer-review-done label.

**Comment by bmcelvee** [comment]
LGTM!

**Review by Srivaralakshmi — APPROVED** [review]

**Pull request merged**
